compute.tf

```diff
@@ -9,10 +9,12 @@ resource "azurerm_windows_virtual_machine" "main" {
   admin_username        = var.local_admin_username
   admin_password        = local.local_admin_password[count.index]
 
+  zone                = var.use_availability_zones ? count.index % 3 + 1 : null
```

Comment:
There is a risk that multiple VMs within the same zone will be deployed within the same fault and/or update domain. I suggest the module instead manages an availability set.

Reply:
Are you sure? I don't think it makes sense to talk about fault domain when we aren't using availability sets. My understanding is that a fault domain is a grouping of VMs within an availability set that shares power and network switching - essentially a rack.
>Fault domains define the group of virtual machines that share a common power source and network switch.

https://docs.microsoft.com/en-us/azure/virtual-machines/availability-set-overview

Whereas an availability zone is essentially its own datacenter:
>Each zone is composed of one or more datacenters equipped with independent power, cooling, and networking infrastructure.

https://docs.microsoft.com/en-us/azure/availability-zones/az-overview

Reply:
My perspective is that using availability zones is the better pattern as it gives control of the availability settings. 
It makes more sense to manage the availability set within the module as the module itself compromises all the hosts that will run the "application".

Regarding the noise, I suggest setting default values for `platform_update_domain_count` and `platform_update_domain_count`. Its your decision whether that constitutes as too much noise or not.